### ISSUE TITLE
broken build from 2.1.0 -> 2.2.0 minor version upgrade

### ISSUE THREAD

**Issue by jab** [PyCQA/astroid#651]
I have a daily cron job in CI that runs pylint, pinned to pylint version 2.2.2. This runs using Python 3.7.

Yesterday this job [succeeded](https://travis-ci.org/jab/bidict/jobs/498784114) but today it [failed](https://travis-ci.org/jab/bidict/jobs/499334076). I changed nothing in between.

Checking the `pip list` output, the only thing that did change is transitively picking up astroid 2.2.0 instead of 2.1.0 (via my direct dependency on pylint).

A minor version upgrade should not cause breaking changes, right?

**Comment by jab** [comment]
All the new lines pylint started flagging after picking up astroid 2.2.2 are spurious.

The lines that look like
```
bidict/_orderedbase.py:191:14: E1101: Instance of 'dict' has no 'inverse' member (no-member)
```
are all due to incorrectly inferring a type of `dict` instance of `bidict`.

Besides that issue, there are many lines like
```
internal error with sending report for module ['...']
'Import' object has no attribute 'infer_name_module'
`

**Comment by PCManticore** [comment]
Easy there with the patronizing tone. I don't owe you anything and especially I don't owe you a green CI. 
It's possible with every release to have new warnings and checks surfaced if `astroid` gains new capabilities or if its inference engine gets improvements, it's also possible that it might have new false positives, in which case I recommend you to submit the issue to pylint.
Regarding the other error, `pylint` was using an unofficial API of `astroid`, which was addressed on pylint's side.

**Comment by jab** [comment]
Had no intention of using a patronizing tone, sorry it came off that way. The "right?" part of "A minor version upgrade should not cause breaking changes, right?" was a genuine question. I wasn't sure if this project intends to conform to semver. When you say "It's possible with every release to have new warnings and checks surfaced...", do you mean with every minor release? Patch release? Just trying to understand the contract so users can pin accordingly.

Thanks for your work on this project.

**Comment by schrodervictor** [comment]
This issue also broke everything on our side. Successful builds of one day became a cascade of failing builds on the next day with no changes introduced.

I can't see any "patronizing " tone on @jab 's comment, but only a concerned and legit feedback from a community user, reporting a problem. This issue should not have been closed so quickly without further investigation.

I would suggest re-opening it.

And thanks for all the awesome work here! We are very grateful that such great tool was developed and made available for everybody!

**Comment by PCManticore** [comment]
Hey folks, maybe I was a bit too harsh in the interpretation of @jab 's message. I left a comment in https://github.com/PyCQA/pylint/pull/2774#issuecomment-469020789 describing the situation with `semver` and a potential fix for 2.2.

As a summary, we don't strictly follow `semver`, as a rule of thumb it should always be latest `pylint` + latest `astroid` for getting the most out of `pylint`.  Also `pylint==2.2.3` limits the `astroid` version it needs to be `astroid<2.2` which should avoid the `infer_name_module` crash. And lastly, while `astroid` is a transitive dependency of `pylint`, it's probably best to pin both of them, especially since minor releases of `astroid` can affect what gets emitted or not on pylint's side.
